### ISSUE TITLE
Change new `np` submodule name to the full `numpy` name 

### ISSUE THREAD

**Issue by godaygo** [numba/numba#5467]
## Feature request 

Maybe it is too late for this type of request... Did you consider to name new `np` module with the full `numpy` name? I understand that this is rather a stylistic cause. Perhaps in the future overloads for various libraries will be contained in subpackages like `numba.numpy`, `numba.scipy` or `numba.sparse` and since the huge refactoring is (really was) going on during 0.49 it will be nice to do it once... I understand that these are trifles, but still.

**Comment by esc** [comment]
@godaygo thanks for this suggestion, we will take it into consideration.

**Comment by stuartarchibald** [comment]
I don't think that Numba would necessarily hold overload implementations for SciPy or Sparse, these should probably live in extension packages like e.g. https://github.com/numba/numba-scipy

The reason it's called `np` internally is due to refactoring tools getting woefully confused by `numpy` the package and a `numba.numpy` submodule. I guess now the moves have largely been done we could change it again but I'm not sure what it gains. Any views @sklam @seibert @esc @DrTodd13 @ehsantn ?

**Comment by ehsantn** [comment]
I don't have a strong opinion either way, but I think making sure tools and IDEs don't get confused is important (I use VS Code which is very useful when it's not confused).

**Comment by seibert** [comment]
Yeah, I think that not matching oddly might be preferable, and I agree that beyond the core NumPy/Python stdlib support, we want to have code for other packages live in extension modules.  (TBH, we might even consider moving NumPy support out of Numba into its own extension as Numba matures further....)

**Comment by godaygo** [comment]
@stuartarchibald Apparently, I did not quite correctly put it. I meant something by analogy with python's namespace packaging, when several different projects for example `numba`, `numba-scipy`, possibly `numba-numpy` all live under one package `numba`. They all can be developed separately like currently is done with `numba-scipy` but after installation the provided by them (by extension package) additional functionality will be accessed through `numba.scipy` or injected during their initialization...  

One of the reasons why I opened this issue was that I just noticed that after refactoring a new sub-module with a jargon name `np` appeared and I thought that this could be an oversight, because every other sub-module has quite adequate name like `cpython`, `datamodel`, ... and because there is no clear understanding of what is public and what is not, this abbreviated name will be settled down in the future.

As I said before this is rather a stylistic cause and possibly an oversight during refactoring. But as I see now that this was done intentionally, this issue can be closed.
